packages/syft/src/syft/core/tensor/lazy_repeat_array.py

```diff
@@ -262,6 +272,19 @@ def sum(self, *args: Tuple[Any, ...], **kwargs: Any) -> lazyrepeatarray:
         res = np.array(self.to_numpy().sum(*args, **kwargs))
         return lazyrepeatarray(data=res, shape=res.shape)
 
+    def mean(self, *args: Tuple[Any, ...], **kwargs: Any) -> lazyrepeatarray:
+        res = np.array(self.to_numpy().mean(*args, **kwargs))
+        data_type = "none"
+        if res.size > 0:
+            all_min_val = np.all(res == res.min())
+            all_max_val = np.all(res == res.max())
+
+        if all_min_val:
+            data_type = "min_val"
+        elif all_max_val:
+            data_type = "max_val"
+        return lazyrepeatarray(data=res, shape=res.shape, data_type=data_type)
```

Comment:
I believe we discussed this during our 1:1- let's make use of the fact that we can predict the output ahead of time, so we can avoid the potentially expensive `self.to_numpy()` operation.